dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -84,69 +89,54 @@
  * @author Lars Helge Overland
  */
 @Slf4j
+@RequiredArgsConstructor
 @Component( "org.hisp.dhis.common.IdentifiableObjectManager" )
-public class DefaultIdentifiableObjectManager
-    implements IdentifiableObjectManager
+public class DefaultIdentifiableObjectManager implements IdentifiableObjectManager
 {
     public static final String DEFAULT = "default";
 
     /**
-     * Cache for default category objects. Disabled during test phase.
+     * "Cache" for default category objects IDs. The IDs never change but the
+     * objects might. The cache exists to avoid doing a lookup by name by
+     * remembering the database ID, so it can be used in subsequent lookups.
      */
-    private final Cache<Long> defaultObjectCache;
+    private final Map<Class<?>, Long> defaultObjectCache = new ConcurrentHashMap<>( 4 );
```

Comment:
FYI: Using an actual `Cache` to cache 4 longs by class name which also would never expire (become invalid or change) was unnecessarily complicated so I used a simple map instead.